Sources/MyStyles/Theme.swift

```diff
@@ -0,0 +1,271 @@
+import SwiftUI
+
+private struct ThemeKey: EnvironmentKey {
+    static let defaultValue: Themable = Theme.current
+}
+
+public extension EnvironmentValues {
+    var theme: Themable {
+        get { self[ThemeKey.self] }
+        set { self[ThemeKey.self] = newValue }
+    }
+}
+
+public enum ThemeColor {
+    case surface, background
+    case content, contentSubtle, contentDisabled, contentSuccess, contentCritical
+    case border, divider
+    
+    case actionPrimary, actionPrimaryBold
+    case actionSecondary, actionDisabled, actionIcon
+    
+    case rating9, rating8, rating7, rating6, rating5
+}
+
+public protocol Themable {
+    // Surfaces
+    func surface(for colorScheme: ColorScheme) -> MyColor
+    func background(for colorScheme: ColorScheme) -> MyColor
+    
+    // Content
+    func content(for colorScheme: ColorScheme) -> MyColor
+    func contentSubtle(for colorScheme: ColorScheme) -> MyColor
+    func contentDisabled(for colorScheme: ColorScheme) -> MyColor
+    func contentSuccess(for colorScheme: ColorScheme) -> MyColor
+    func contentCritical(for colorScheme: ColorScheme) -> MyColor
+    func border(for colorScheme: ColorScheme) -> MyColor
+    func divider(for colorScheme: ColorScheme) -> MyColor
+    
+    // Actions
+    func actionPrimary(for colorScheme: ColorScheme) -> MyColor
+    func actionPrimaryBold(for colorScheme: ColorScheme) -> MyColor
+    func actionSecondary(for colorScheme: ColorScheme) -> MyColor
+    func actionDisabled(for colorScheme: ColorScheme) -> MyColor
+    func actionIcon(for colorScheme: ColorScheme) -> MyColor
+    
+    // Ratings
+    func rating9(for colorScheme: ColorScheme) -> MyColor
+    func rating8(for colorScheme: ColorScheme) -> MyColor
+    func rating7(for colorScheme: ColorScheme) -> MyColor
+    func rating6(for colorScheme: ColorScheme) -> MyColor
+    func rating5(for colorScheme: ColorScheme) -> MyColor
+    
+    // General
+    func color(for themeColor: ThemeColor, colorScheme: ColorScheme) -> MyColor
+}
+
+extension Themable {
+    func color(for themeColor: ThemeColor, colorScheme: ColorScheme) -> MyColor {
```

Comment:
It seems a shame to lose the ability for the view to access the colors by the static vars.

```
Circle()
    .foregroundColor(.actionIcon)
```
vs
```
Circle()
    .foregroundColor(theme.color(for: .actionIcon, colorScheme: .light)
```

I think @maloneranger  had talked about some way of registering the primitive tokens that the functional tokens point to by way of some sort of `ThemeEngine` class?